packages/legend-query-builder/src/components/__tests__/QueryBuilderVariables.test.tsx

```diff
@@ -0,0 +1,228 @@
+      'model::MyRuntime',
+      TEST_DATA__ModelCoverageAnalysisResult_ComplexRelational,
+    );
+    await act(async () => {
+      queryBuilderState.initializeWithQuery(
+        create_RawLambda(
+          TEST_DATA__simpleProjection.parameters,
+          TEST_DATA__simpleProjection.body,
+        ),
+      );
+      queryBuilderState.setShowParametersPanel(true);
+      queryBuilderState.constantState.setShowConstantPanel(true);
+    });
+
+    await waitFor(() =>
+      renderResult.getByTestId(QUERY_BUILDER_TEST_ID.QUERY_BUILDER_PARAMETERS),
+    );
+    const parametersPanel = renderResult.getByTestId(
+      QUERY_BUILDER_TEST_ID.QUERY_BUILDER_PARAMETERS,
+    );
+
+    fireEvent.click(getByTitle(parametersPanel, 'Add Parameter'));
+
+    await waitFor(() => renderResult.getByText('Create'));
+
+    fireEvent.click(renderResult.getByText('Create'));
+
+    await waitFor(() =>
+      renderResult.getByTestId(QUERY_BUILDER_TEST_ID.QUERY_BUILDER_CONSTANTS),
+    );
+    const constantsPanel = renderResult.getByTestId(
+      QUERY_BUILDER_TEST_ID.QUERY_BUILDER_CONSTANTS,
+    );
+
+    fireEvent.click(getByTitle(constantsPanel, 'Add Constant'));
+
+    await act(async () => {
+      if (!queryBuilderState.constantState.selectedConstant) {
+        return;
+      }
+
+      const varExpression =
+        queryBuilderState.constantState.selectedConstant.variable;
+      variableExpression_setName(varExpression, 'var_1');
+    });
+
+    expect(
+      await waitFor(() =>
+        renderResult.getByText('Constant Name Already Exists'),
+      ),
+    ).not.toBeNull();
+
+    expect(renderResult.getByText('Create').hasAttribute('disabled')).toBe(
+      true,
+    );
+  },
+);
+
+test(
+  integrationTest(
+    'Query builder shows humanized date label value for constant variable with type Date',
+  ),
+  async () => {
+    const { renderResult, queryBuilderState } = await TEST__setUpQueryBuilder(
+      TEST_DATA__ComplexRelationalModel,
+      stub_RawLambda(),
+      'model::relational::tests::simpleRelationalMapping',
+      'model::MyRuntime',
+      TEST_DATA__ModelCoverageAnalysisResult_ComplexRelational,
+    );
+    await act(async () => {
+      queryBuilderState.initializeWithQuery(
+        create_RawLambda(
+          TEST_DATA__simpleProjection.parameters,
+          TEST_DATA__simpleProjection.body,
+        ),
+      );
+      queryBuilderState.constantState.setShowConstantPanel(true);
+    });
+
+    await waitFor(() =>
+      renderResult.getByTestId(QUERY_BUILDER_TEST_ID.QUERY_BUILDER_CONSTANTS),
+    );
+    const constantsPanel = renderResult.getByTestId(
+      QUERY_BUILDER_TEST_ID.QUERY_BUILDER_CONSTANTS,
+    );
+    fireEvent.click(getByTitle(constantsPanel, 'Add Constant'));
+
+    await act(async () => {
+      if (!queryBuilderState.constantState.selectedConstant) {
+        return;
+      }
+      queryBuilderState.constantState.selectedConstant.changeValSpecType(
```

Comment:
can we pick the strict date as a user would from the dropdown.

Reply:
I believe, same issue with accessing dropdown text